File: 01_CONTAINS_DUPLICATES/Solution.java
import java.util.Arrays;
import java.util.Set;
import java.util.HashSet;

public class Solution {

    

    // public boolean soluction01(int[] nums) {
    //     for (int i = 0; i < nums.length; i++) {
    //         for (int j = i + 1; j < nums.length; j++) {
    //             if (nums[i] == nums[j]) {
    //                 return true;
    //             }
    //         }
    //     }
    //     return false;
    // }

    // public boolean soluction02(int[] nums){
    //     Arrays.sort(nums);
    //     for (int i = 0;i<nums.length; i++) {
    //         if (nums[i] == nums[i-1]){
    //             return true;
    //         }
    //     }
    //     return false;
    // }

    // public boolean soluction03(int[] nums) {
    //     Set<Integer> seen = new HashSet<>();

    //     for (int num: nums){
    //         if (seen.contains(nums)){
    //             return true;
    //         }
    //         seen.add(num);
    //     }
    //     return false;

    // }

    // public boolean soluction04(int[] nums) {
    //     return Arrays.stream(nums).distinct().count() < nums.length;
    // }


    public static void main(String args[]) {}
}

File: 01_CONTAINS_DUPLICATES/Solution.py
from typing import List

class Soluction:
   
   def hasDuplicate(self, nums:List[int]) -> bool:
        for i in range(len(nums)):
            if nums[i] in nums[i+1:]:
                return True 
            return False


    # def soluction_01(self, nums:List[int]) -> bool:
    #     for i in range(len(nums)):
    #         for j in range(i + 1, len(nums)):
    #             if nums[i] == nums[j]:
    #                 return True
    #     return False



    # def soluction_02(self, nums:List[int]) -> bool :
    #     nums.sort()
    #     for i in range(1, len(nums)):
    #         if nums[i] == nums[i - 1] :
    #             return True
    #     return False


    # def soluction_03(self, nums:List[int]) -> bool :
    #     seen = set()
    #     for num in nums:
    #         if num in seen:
    #             return True 
    #         seen.add(num)
    #     return False


    # def soluction_04(self, nums:List[int]) -> bool:
    #     return len(set(nums)) < len(nums)